packages/plugin-edgedb/src/edgedb-field-builder.ts

```diff
@@ -0,0 +1,189 @@
+import { GraphQLResolveInfo } from 'graphql';
+import {
+  CompatibleTypes,
+  FieldKind,
+  FieldRef,
+  InputFieldMap,
+  NormalizeArgs,
+  RootFieldBuilder,
+  SchemaTypes,
+  ShapeFromTypeParam,
+  TypeParam,
+} from '@pothos/core';
+import { FieldMap } from './util/relation-map';
+import { getLink, getRefFromModel } from './util/datamodel';
+import { extractTargetTypeName } from './util/target';
+import { isMultiLink } from './util/links';
+import { RelatedFieldOptions } from './types';
+
+// Workaround for FieldKind not being extended on Builder classes
+const RootBuilder: {
+  // eslint-disable-next-line @typescript-eslint/prefer-function-type
+  new <Types extends SchemaTypes, Shape, Kind extends FieldKind>(
+    name: string,
+    builder: PothosSchemaTypes.SchemaBuilder<Types>,
+    kind: FieldKind,
+    graphqlKind: PothosSchemaTypes.PothosKindToGraphQLType[FieldKind],
+  ): PothosSchemaTypes.RootFieldBuilder<Types, Shape, Kind>;
+} = RootFieldBuilder as never;
+
+export class EdgeDBObjectFieldBuilder<
+  Types extends SchemaTypes,
+  Model extends
+    | ({ [ModelKey in keyof Model]: Model[ModelKey] extends infer U ? U : never } & {
+        Fields: string | never;
+        Links: {
+          [Key in Model['Fields']]: { Shape: Model['Links'][Key] };
```

Comment:
curious about this shape referencing itself here, what is that used for?

Reply:
I had parent args, Fields and Links in one type which end up really nested. Will need to rethink that and construct it in a clean way, will probably refer more or less to `PrismaTypes`'s structure. This Shape referencing isnt actually useful there indeed.

https://github.com/hayes/pothos/blob/1f9bec0ed08065585a2c35f9e2afa8bc93cc60a4/packages/plugin-prisma/src/types.ts#L37-L53

Curios is there a difference between `Fields` and `RelationName`? 
Looks like they contain the same values.

Reply:
Nope... That seems to be an oversight.  I'll remove `Fields` from the prisma plugin